### ISSUE TITLE
Suddenly sensitive resource attributes are exposed by "terraform show" in latest Terraform version 1.4.x

### ISSUE THREAD

**Issue by Nantero1** [confluentinc/terraform-provider-confluent#238]
### Terraform Version

```shell
Terraform v1.4.5
on windows_amd64
+ provider registry.terraform.io/confluentinc/confluent v1.36.0
+ provider registry.terraform.io/hashicorp/azuread v2.37.1
+ provider registry.terraform.io/hashicorp/azurerm v3.51.0
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = ">= 1.3.2"

  backend "azurerm" {}

  required_providers {
    confluent = {
      source  = "confluentinc/confluent"
      version = "~> 1.36.0"
    }
  }
}
```

### Debug Output

I am sorry, I think it is not needed at the moment. Please let me know if you think debug output is important.

### Expected Behavior
Like in previous terraform versions (tested with 1.3.9), resource attributes marked as `sensitive` by a provider shall be not exposed in the `terraform show` statement, but masked behind a `(sensitive value)` placeholder.

### Actual Behavior

There was a change in behaviour, in Terraform 1.3.9 resource attributes marked as "sensitive" by a provider were masked by the text ` (sensitive value)`. Now - in Terraform 1.4.5 - these values are exposed and can be seen in clear text. No change of provider version happened, the only thing changed was the used Terraform version. 

Please see the attached screenshot. On the left hand side the output of Terraform 1.4.5 (values exposed), on the right hand side the output of Terraform 1.3.9 (sensitive values masked). Same code, same provider versions, only the Terraform version changed.

![grafik](https://user-images.githubusercontent.com/12953323/233322414-947f7de4-9b2b-40ff-9eba-1289883a0e8e.png)

Tested all version starting from 1.4.0. All versions starting with 1.4.0 show this behaviour. The breaking change happened between version 1.3.9 and 1.4.0.

[The documentation reads, that](https://developer.hashicorp.com/terraform/cli/commands/show):

> Note: When using the -json command-line flag, any sensitive values in Terraform state will be displayed in plain text. For more information, see [Sensitive Data in State](https://developer.hashicorp.com/terraform/language/state/sensitive-data).

But this contradicts the above described behaviour... I see sensitive values even in non-json terraform show output.

### Steps to Reproduce

1. Use this terraform module to create an example resource with sensitive values inside, f.e. `confluent_api_key`
2. Run "terraform init" and "terraform apply"
3. Observe the output of `terraform show`, sensitive values are exposed in latest terraform version.

**Comment by felixboehm** [comment]
This provider flags the attribute of the resource used in the example above as sensitive:

https://github.com/confluentinc/terraform-provider-confluent/blob/master/internal/provider/resource_api_key.go#L85

Probably an issue in Terraform.

**Comment by Nantero1** [comment]
Terraform just confirmed, that this is an issue with Terraform v1.4.0 - v1.4.6. A fix of Terraform shall be provided next week. https://github.com/hashicorp/terraform/issues/33055#issuecomment-1516184252

**Comment by linouk23** [comment]
Thanks for asking @Nantero1 and thanks to all of you folks for the investigation!

Let's wait until [the fix](https://github.com/hashicorp/terraform/issues/33055#issuecomment-1516184252) from HashiCorp is in and then it might be a good time to close this issue.

**Comment by felixboehm** [comment]
Solved with tf 1.4.6
https://github.com/hashicorp/terraform/pull/32629/files
